### ISSUE TITLE
Compilation failture on Windows

### ISSUE THREAD

**Issue by Kixiron** [maps4print/azul#213]
## Description
Building azul fails with the following error
```
error: failed to run custom build command for `harfbuzz-sys v0.3.0 (https://github.com/maps4print/azul-dependencies?rev=bf2933b9aac43a7003278862772250398e4fa457#bf2933b9)`

Caused by:
  process didn't exit successfully: `G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-4d7deefbb1bf663c\build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=HARFBUZZ_SYS_NO_PKG_CONFIG
running: "cmake" "C:\\Users\\Chase\\.cargo\\git\\checkouts\\azul-dependencies-70bb1f94316762f9\\bf2933b\\harfbuzz-sys-0.3.0\\harfbuzz" "-G" "Visual Studio 14 2015 Win64" "-Thost=x64" "-DCMAKE_INSTALL_PREFIX=G:\\LocalFileServer\\Programming\\Rust\\discordant\\target\\debug\\build\\harfbuzz-sys-fdc4c0a27e8872c4\\out" "-DCMAKE_C_FLAGS= /nologo /MD" "-DCMAKE_C_FLAGS_DEBUG= /nologo /MD" "-DCMAKE_CXX_FLAGS= /nologo /MD" "-DCMAKE_CXX_FLAGS_DEBUG= /nologo /MD" "-DCMAKE_BUILD_TYPE=Debug"
-- Selecting Windows SDK version  to target Windows 10.0.18362.
-- Configuring incomplete, errors occurred!
See also "G:/LocalFileServer/Programming/Rust/discordant/target/debug/build/harfbuzz-sys-fdc4c0a27e8872c4/out/build/CMakeFiles/CMakeOutput.log".

--- stderr
CMake Error at CMakeLists.txt:2 (project):
  Failed to run MSBuild command:

    C:/Program Files (x86)/MSBuild/14.0/bin/MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 14.0.25420.1
    Copyright (C) Microsoft Corporation. All rights reserved.

    Build started 8/26/2019 2:05:09 PM.
    Project "G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj" on node 1 (default targets).
    G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj(14,2): error MSB4019: The imported project "G:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    Done Building Project "G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj" (default targets) -- FAILED.

    Build FAILED.

    "G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj" (default target) (1) ->
      G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj(14,2): error MSB4019: The imported project "G:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

        0 Warning(s)
        1 Error(s)

    Time Elapsed 00:00:00.08


  Exit code: 1



thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', C:\Users\Chase\.cargo\git\checkouts\azul-dependencies-70bb1f94316762f9\bf2933b\cmake-0.1.35\src\lib.rs:778:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

## Version / OS

* azul version:
Current git, commit #6da27d0

* Operating system:
Windows 10 Pro

## Steps to Reproduce
Attempt to build azul

## Additional Information
Using `RUST_BACKTRACE=1`, the following output is produced
```
error: failed to run custom build command for `harfbuzz-sys v0.3.0 (https://github.com/maps4print/azul-dependencies?rev=bf2933b9aac43a7003278862772250398e4fa457#bf2933b9)`

Caused by:
  process didn't exit successfully: `G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-4d7deefbb1bf663c\build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=HARFBUZZ_SYS_NO_PKG_CONFIG
running: "cmake" "C:\\Users\\Chase\\.cargo\\git\\checkouts\\azul-dependencies-70bb1f94316762f9\\bf2933b\\harfbuzz-sys-0.3.0\\harfbuzz" "-G" "Visual Studio 14 2015 Win64" "-Thost=x64" "-DCMAKE_INSTALL_PREFIX=G:\\LocalFileServer\\Programming\\Rust\\discordant\\target\\debug\\build\\harfbuzz-sys-fdc4c0a27e8872c4\\out" "-DCMAKE_C_FLAGS= /nologo /MD" "-DCMAKE_C_FLAGS_DEBUG= /nologo /MD" "-DCMAKE_CXX_FLAGS= /nologo /MD" "-DCMAKE_CXX_FLAGS_DEBUG= /nologo /MD" "-DCMAKE_BUILD_TYPE=Debug"
-- Selecting Windows SDK version  to target Windows 10.0.18362.
-- Configuring incomplete, errors occurred!
See also "G:/LocalFileServer/Programming/Rust/discordant/target/debug/build/harfbuzz-sys-fdc4c0a27e8872c4/out/build/CMakeFiles/CMakeOutput.log".

--- stderr
CMake Error at CMakeLists.txt:2 (project):
  Failed to run MSBuild command:

    C:/Program Files (x86)/MSBuild/14.0/bin/MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 14.0.25420.1
    Copyright (C) Microsoft Corporation. All rights reserved.

    Build started 8/26/2019 2:06:05 PM.
    Project "G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj" on node 1 (default targets).
    G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj(14,2): error MSB4019: The imported project "G:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    Done Building Project "G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj" (default targets) -- FAILED.

    Build FAILED.

    "G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj" (default target) (1) ->
      G:\LocalFileServer\Programming\Rust\discordant\target\debug\build\harfbuzz-sys-fdc4c0a27e8872c4\out\build\CMakeFiles\3.15.2\VCTargetsPath.vcxproj(14,2): error MSB4019: The imported project "G:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

        0 Warning(s)
        1 Error(s)

    Time Elapsed 00:00:00.04


  Exit code: 1



thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', C:\Users\Chase\.cargo\git\checkouts\azul-dependencies-70bb1f94316762f9\bf2933b\cmake-0.1.35\src\lib.rs:778:5
stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized
             at C:\Users\VssAdministrator\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.29\src\backtrace\mod.rs:66
   1: std::sys_common::backtrace::_print
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\sys_common\backtrace.rs:47
   2: std::sys_common::backtrace::print
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\sys_common\backtrace.rs:36
   3: std::panicking::default_hook::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:200
   4: std::panicking::default_hook
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:214
   5: std::panicking::rust_panic_with_hook
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:477
   6: std::panicking::continue_panic_fmt
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:384
   7: std::panicking::begin_panic_fmt
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:339
   8: cmake::fail
             at C:\Users\Chase\.cargo\git\checkouts\azul-dependencies-70bb1f94316762f9\bf2933b\cmake-0.1.35\src\lib.rs:778
   9: cmake::run
             at C:\Users\Chase\.cargo\git\checkouts\azul-dependencies-70bb1f94316762f9\bf2933b\cmake-0.1.35\src\lib.rs:759
  10: cmake::Config::build
             at C:\Users\Chase\.cargo\git\checkouts\azul-dependencies-70bb1f94316762f9\bf2933b\cmake-0.1.35\src\lib.rs:611
  11: build_script_build::main
             at .\build.rs:21
  12: std::rt::lang_start::{{closure}}<()>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libstd\rt.rs:64
  13: std::rt::lang_start_internal::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\rt.rs:49
  14: std::panicking::try::do_call<closure,i32>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:296
  15: panic_unwind::__rust_maybe_catch_panic
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libpanic_unwind\lib.rs:82
  16: std::panicking::try
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:275
  17: std::panic::catch_unwind
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panic.rs:394
  18: std::rt::lang_start_internal
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\rt.rs:48
  19: std::rt::lang_start<()>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libstd\rt.rs:64
  20: main
  21: invoke_main
             at d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  22: __scrt_common_main_seh
             at d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  23: BaseThreadInitThunk
  24: RtlUserThreadStart

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by sandouli** [comment]
The error message "The imported project "G:\Microsoft.Cpp.Default.props" was not found." led me to this page https://stackoverflow.com/questions/41695251/c-microsoft-cpp-default-props-was-not-found.

Could you try installing the Windows build tools and setting the `VCTargetsPath` environment variable as instructed ?

**Comment by RichardBenes** [comment]
I am experiencing this issue too, on rustc and cargo 1.38.0 installed with MSVC, Win10 v1903 64bit.

**Comment by ConnorBP** [comment]
Getting this issue as well. I beleive my visual studio version (2019) is too new for the harfbuzz-sys dependency for whatever reason.

**Comment by Heavenston** [comment]
Same issue !

**Comment by fschutt** [comment]
This seems to be the same issue as #203 which should be fixed (temporarily) thanks to #236. I can't test it, just reopen this issue if it still exists.
